File: Year 2/Semester 4/Programming and Projection Environments/Week 7/SeventhWeekServerClient/Persistance/src/main/java/SeventhWeekServerClient/persistance/repository/IShowRepository.java
package SeventhWeekServerClient.persistance.repository;

import SeventhWeekServerClient.domain.Entity;
import SeventhWeekServerClient.domain.ShowDTO;

public interface IShowRepository <ID, E extends Entity<ID>> extends IRepository<ID, E> {
    Iterable<ShowDTO> getAllShowsDTO();
}

File: Year 2/Semester 4/Programming and Projection Environments/Week 8/SeventhWeekCSharpServerClient/persistence/interfaces/IShowRepository.cs
using SeventhWeekCSharpServerClient.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SeventhWeekCSharpServerClient.repository.interfaces
{
    public interface IShowRepository : IRepository<int, Show<int>>
    {
        List<ShowDTO<int>> GetAllShows();
    }
}
